hydra/utils.py

```diff
@@ -33,6 +34,36 @@ class ConvertMode(Enum):
     ALL = 2
 
 
+def unbox(obj: Any, convert: ConvertMode):
+    """Unbox object."""
+    if isinstance(obj, Boxed):
+        result = unbox(obj.obj, convert)
+    elif isinstance(obj, dict):
+        result = {key: unbox(value, convert) for key, value in obj.items()}
+    elif OmegaConf.is_dict(obj):
+        d = {} if convert == ConvertMode.ALL else OmegaConf.create(flags={"allow_objects": True})
+        for key, value in obj.items():
+            d[key] = unbox(value, convert)
+        result = d
+    elif isinstance(obj, list):
+        result = [unbox(item, convert) for item in obj]
+    elif OmegaConf.is_list(obj):
+        lst = [] if convert == ConvertMode.ALL else OmegaConf.create([], flags={"allow_objects": True})
+        for it in obj:
+            lst.append(unbox(it, convert))
+        result = lst
+    elif convert != ConvertMode.ALL:
```

Comment:
I add this for the PARTIAL mode -> I take the data class instance and convert it back to a structured config